web/src/resources/osk/kmwosk.css

```diff
@@ -555,10 +550,33 @@ div.android div.kmw-keytip-cap {
 .phone.android #kmw-popup-keys {border:none; border-radius: 2px; background-color:#333333; padding:5px 5px 0 0;}
 .tablet.android #kmw-popup-keys {border:1px solid #eee; border-radius: 3px; background-color:#888; padding:8px 12px 4px 4px;}
 
+.kmw-embedded .phone.ios #kmw-popup-keys,
+.kmw-embedded .tablet.ios #kmw-popup-keys,
+.kmw-embedded .phone.android #kmw-popup-keys {
+  padding-bottom: 4px;
+}
+.kmw-embedded .tablet.android #kmw-popup-keys {
+  padding-bottom: 8px;
+}
```

Comment:
The important part for #7890.

Most of this commit's other changes are actually in support of #7889; I'd neglected to comprehensively specify the styling.

----

I _do_ find it concerning that I don't yet know _why_ the extra padding must be explicitly specified when within a WebView but not when outside a webview context.  It does seem to hold consistently, though.

Reply:
Possible lead:  I can repro the effect in-browser if I set a specific height on the `kmw-popup-keys` element (the subkey container) equal to the height of the subkeys.  That said... there's no such spec within the Android app on a phone form-factor device, yet #7890 is still an issue there.